src/main/kotlin/com/chloe/kotlinserv/MainVerticle.kt

```diff
@@ -1,23 +1,97 @@
 package com.chloe.kotlinserv
 
 import com.google.gson.Gson
-import io.vertx.core.http.HttpServerResponse
-import io.vertx.ext.web.Router
+import com.typesafe.config.ConfigFactory
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import java.io.File
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 
 fun main(args: Array<String>) {
+    val file = File(args[0])
+    val config = ConfigFactory.parseFile(file)
+
+    val url = config.getString("clickhouse.url")
+
+    val conf = HikariConfig()
+    conf.jdbcUrl = url
+    conf.driverClassName = "com.clickhouse.jdbc.ClickHouseDriver"
+    conf.username = config.getString("clickhouse.user")
+    conf.password = config.getString("clickhouse.password")
+
+    val ds = HikariDataSource(conf)
+
+    val port = config.getInt("port")
     val myServer = VertxHttpServer()
     val json = Gson()
 
     val getRoute = HttpRoute("/countrystats", HttpMethod.GET) {
-        val data = CountryStats("23-12-2000", "RUS", 3)
-        HttpResponse(200, responseBody=json.toJson(data), mapOf("content-type" to "application/json"))
+        val connection = ds.connection
+
+        connection.use {
+            val statement =
+                connection.prepareStatement("SELECT country, toDate(timestamp) as time, count(country) FROM chloe.events WHERE country='user' group by country, toDate(timestamp)")
+            val result = statement.executeQuery()
+
+            val list = mutableListOf<CountryStats>()
+            while (result.next()) {
+                list.add(
+                    CountryStats(
+                        data = result.getString(1),
+                        country = result.getString(2),
+                        count = result.getInt(3)
+                    )
+                )
+            }
+
+            if (list.isEmpty()) {
+                HttpResponse(
+                    code = 204,
```

Comment:
http code 204 indicate No Content by default, so response body should be always null